converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Observation/ObservationParser.cs

```diff
@@ -6,59 +6,48 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
+using System.Linq;
 using EnsureThat;
 using FellowOakDicom;
 using FellowOakDicom.StructuredReport;
 using Hl7.Fhir.Model;
 
 namespace Microsoft.Health.DicomCast.Core.Features.Worker.FhirTransaction;
 
-public class ObservationParser
+public static class ObservationParser
 {
-    public IReadOnlyCollection<Observation> Parse(DicomDataset dataset, ResourceReference patientReference, ResourceReference imagingStudyReference, Identifier identifier)
+    public static IReadOnlyCollection<Observation> Parse(DicomDataset dataset, ResourceReference patientReference, ResourceReference imagingStudyReference, Identifier identifier)
     {
         EnsureArg.IsNotNull(dataset, nameof(dataset));
         EnsureArg.IsNotNull(patientReference, nameof(patientReference));
         EnsureArg.IsNotNull(imagingStudyReference, nameof(imagingStudyReference));
         EnsureArg.IsNotNull(identifier, nameof(identifier));
 
-        var observations = new List<Observation>();
-        ParseDicomDataset(dataset, patientReference, imagingStudyReference, observations, identifier);
-
-        return observations;
+        return ParseDicomDataset(dataset, patientReference, imagingStudyReference, identifier).ToList();
     }
 
-    private void ParseDicomDataset(DicomDataset dataset, ResourceReference patientReference, ResourceReference imagingStudyReference, List<Observation> observations, Identifier identifier)
+    private static IEnumerable<Observation> ParseDicomDataset(DicomDataset dataset, ResourceReference patientReference, ResourceReference imagingStudyReference, Identifier identifier)
```

Comment:
@brandonpollett / @KnicKnic - I did a more significant refactor here, as I tried to remove the exception-based flow control and avoid passing in the list to mutate it. An analyzer flagged the list parameter